cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/batch/mapreduce/MapReduceSourceContext.java

```diff
@@ -119,10 +117,25 @@ public void setInput(final Input input) {
     LogContext.runWithoutLoggingUnchecked(new Callable<Void>() {
       @Override
       public Void call() throws Exception {
-        Input trackableInput =  ExternalDatasets.makeTrackable(mrContext.getAdmin(), input);
+        externalInput = suffixInput(input);
+        Input trackableInput =  ExternalDatasets.makeTrackable(mrContext.getAdmin(), externalInput);
         mrContext.addInput(trackableInput);
         return null;
       }
     });
   }
+
+  public String getAlias() {
```

Comment:
Ah ok.
But if we have multi input, then there won't be just one `externalInput`, right? There can be multiple then.